### PR TITLE
Add benchmark graph and script for tanh function approximation methods

### PR DESCRIPTION
**Pull request by arturomf94**
## Description
Following https://github.com/OpenMined/PySyft/pull/4031, this PR includes:

- Scripts for plotting benchmark graph for tanh function approximations.
- Benchmark graph as suggested in issue https://github.com/OpenMined/PySyft/issues/3999

Closes: https://github.com/OpenMined/PySyft/issues/3999

## How has this been tested?
- Same as https://github.com/OpenMined/PySyft/pull/4031, no tests are including for benchmarking.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by aanurraj** [comment]
> @aanurraj how does this look to you? Essentially, I recycled your code for the tanh func. Let me know if you notice anything worth changing... 😃

Looks good 👍🏻

**Review by marload — APPROVED** [review]
LGTM 🚀 
I recommend you create a new issue to eliminate this duplication because it duplicate a lot with `benchmark_sigmoid.py`.

File: benchmarks/frameworks/torch/mpc/scripts/benchmark_sample_data.py [hunk 1]
```diff
@@ -1,4 +1,9 @@
-# this is sample data for benchmark of sigmoid function
+"""
+Sample data for the benchmarking of FPT/AST with differnt types of approximations
+See: https://github.com/OpenMined/PySyft/issues/3997
+
+data format: ('method_name', precision value)
```
Comment (Syzygianinfern0) [inline benchmarks/frameworks/torch/mpc/scripts/benchmark_sample_data.py:5]:
```suggestion
data format: list[('method_name', precision value),]
```

Comment (Syzygianinfern0) [inline benchmarks/frameworks/torch/mpc/scripts/benchmark_sample_data.py:5]:
```suggestion
data format: list[('method_name', precision value),]
```

File: benchmarks/frameworks/torch/mpc/scripts/benchmark_tanh.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+def tanh_approximation_plot(benchmark_data_tanh):
+    """
+    This function plots the graph for various sigmoidal approximation benchmarks namely
+    'chebyshev', 'maclaurin', 'exp'.
+
+    Args:
+        benchmark_data_sigmoid (list): the sample data to approximate
+
+    Returns:
+        sigmoid_function_approximations_benchmark (png): plotted graph in graph directory
```
Comment (Syzygianinfern0) [inline benchmarks/frameworks/torch/mpc/scripts/benchmark_tanh.py:47]: Incorrect doc-strings :stuck_out_tongue:

**Review by Syzygianinfern0 — CHANGES_REQUESTED** [review]

File: benchmarks/frameworks/torch/mpc/scripts/benchmark_tanh.py (hunk 2 above)
Comment (arturomf94) [reply benchmarks/frameworks/torch/mpc/scripts/benchmark_tanh.py]: Ah! You're right! Fixed this now.

File: benchmarks/frameworks/torch/mpc/scripts/benchmark_sample_data.py (hunk 1 above)
Comment (arturomf94) [reply benchmarks/frameworks/torch/mpc/scripts/benchmark_sample_data.py]: Ok :) changed.

**Review by Syzygianinfern0 — APPROVED** [review]
Thank you for your contribution @arturomf94 :partying_face: :tada:

**Comment by gmuraru** [comment]
> LGTM 
> I recommend you create a new issue to eliminate this duplication because it duplicate a lot with `benchmark_sigmoid.py`.

Yep, I think this is a great idea! Could you create an issue? :D

**Review by gmuraru — APPROVED** [review]

**Comment by marload** [comment]
> > LGTM
> > I recommend you create a new issue to eliminate this duplication because it duplicate a lot with `benchmark_sigmoid.py`.
> 
> Yep, I think this is a great idea! Could you create an issue? :D

Okay, I will create new issue. 😄

**Pull request merged**